### ISSUE TITLE
False positive error

### ISSUE THREAD

**Issue by shepilov-vladislav** [OCA/pylint-odoo#242]
```
Missing external dependency "odoo" from manifest. More info: https://github.com/OCA/odoo-community.org/blob/master/website/Contribution/CONTRIBUTING.rst#external-dependencies
```

```
Missing external dependency "odoo.addons...." from manifest. More info: https://github.com/OCA/odoo-community.org/blob/master/website/Contribution/CONTRIBUTING.rst#external-dependencies
```

**Comment by pedrobaeza** [comment]
We haven't faced these errors ever. If you don't put in context...

**Comment by moylop260** [comment]
What is your python version, pylint, astroid and pylint-odoo?
Could you share the minimal code to reproduce it, please?

**Comment by blaggacao** [comment]
`odoo` is not in the environment.

**Comment by blaggacao** [comment]
you can reproduce by using this: https://github.com/xoe-labs/dockery-odoo-scaffold
https://xoe-labs.github.io/dockery-odoo/

While it's not minimal, it's the environment to reproduce unless you have `odoo` installed on your host machine.

**Comment by blaggacao** [comment]
@shepilov-vladislav specifically see also: https://github.com/xoe-labs/dockery-odoo-scaffold/blame/0c634238cc0675c166d76ffff06fde4b13efbc62/.pylintrc#L57

**Comment by moylop260** [comment]
@blaggacao 
You are confusing “missing external dependency from manifest” vs “Import Error” are very different.
The first one for pylint-odoo and the second one is a core check from pylint itself.

Then we won’t fix here
And we don’t have enough information to reproduce the first one

**Comment by blaggacao** [comment]
@moylop260
sed '/You are confusing “missing external dependency from manifest” vs “Import Error” are very different.
The first one for pylint-odoo and the second one is a core check from pylint itself./Why do you think they convlut?/'
Would be just so more productive...

But maybe you can explain why do you think this is closeable?

**Comment by blaggacao** [comment]
@moylop260 
> Why do you think they convolut?

I think they convolute, because I can imagine that `odoo` not being in the environment, both checks give false positive resulting in complaining. That would be how they are "not so different".

Based on this maybe you (or @pedrobaeza) can enlighten us if there is a way to trick pyltint *and* `pylint_odoo` into taking on a different environment in a git-commit hook.

Ultimately it's probably a wont-fix, but just maybe you have an idea how we can deal with this... :wink:

**Comment by moylop260** [comment]
Could you check if the following comment fixes this one too, please?
 - https://github.com/OCA/pylint-odoo/issues/284#issuecomment-654390704

Regards!

**Comment by blaggacao** [comment]
For what it's worth, `nix-shell` is one very promising option to generate a predictable environment for this sort of tooling.
